File: CS/Tests/ODataTestsBase.cs
using System;
using System.Diagnostics;
using System.IO;
using Default;
using NUnit.Framework;

namespace Tests {
    public abstract class ODataTestsBase {

        const string ODataServiceUrl = "http://localhost:5000/";
        Process iisProcess = null;

        protected Container GetODataContainer() {
            return new Container(new Uri(ODataServiceUrl));
        }

        [OneTimeSetUp]
        public void OneTimeSetup() {
            string iisExpressPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "IIS Express", "iisexpress.exe");
            string appPath = Path.GetDirectoryName(this.GetType().Assembly.Location);
            appPath = Path.GetFullPath(Path.Combine(appPath, "..", "..", "..", "ODataService"));
            string args = string.Format("/path:\"{0}\" /port:5000", appPath);
            iisProcess = Process.Start(iisExpressPath, args);
        }

        [OneTimeTearDown]
        public void OneTimeTearDown() {
            if(iisProcess != null) {
                iisProcess.Kill();
            }
        }

        [SetUp]
        public void Setup() {
            Container container = GetODataContainer();
            container.InitializeDatabase().Execute();
        }
    }
}

File: VB/Tests/ODataTestsBase.vb
Imports System
Imports System.Diagnostics
Imports System.IO
Imports [Default]
Imports NUnit.Framework

Namespace Tests
    Public MustInherit Class ODataTestsBase

        Private Const ODataServiceUrl As String = "http://localhost:5000/"
        Private iisProcess As Process = Nothing

        Protected Function GetODataContainer() As Container
            Return New Container(New Uri(ODataServiceUrl))
        End Function

        <OneTimeSetUp> _
        Public Sub OneTimeSetup()
            Dim iisExpressPath As String = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "IIS Express", "iisexpress.exe")
            Dim appPath As String = Path.GetDirectoryName(Me.GetType().Assembly.Location)
            appPath = Path.GetFullPath(Path.Combine(appPath, "..", "..", "..", "ODataService"))
            Dim args As String = String.Format("/path:""{0}"" /port:5000", appPath)
            iisProcess = Process.Start(iisExpressPath, args)
        End Sub

        <OneTimeTearDown> _
        Public Sub OneTimeTearDown()
            If iisProcess IsNot Nothing Then
                iisProcess.Kill()
            End If
        End Sub

        <SetUp> _
        Public Sub Setup()
            Dim container As Container = GetODataContainer()
            container.InitializeDatabase().Execute()
        End Sub
    End Class
End Namespace
